pyiron/sphinx/base.py

```diff
@@ -713,128 +984,228 @@ def _set_kpoints(
+        with open(file_name, "w") as f:
+            f.write(f"//{self.job_name}\n")
+            f.write("//SPHInX input file generated by pyiron\n")
+            f.write("format paw;\n")
+            f.write("include <parameters.sx>;\n")
+            f.write("include <userparameters.sx>;\n")
+            f.write("pawPot { include <pawPot.sx>; ")
+            if enable_kjxc:
+                f.write("kjxc; ")
+            f.write("}\n")
+            f.write("structure { include <structure.sx>; }\n")
+            f.write("basis { include <basis.sx>; }\n")
+            f.write("PAWHamiltonian { include <PAWHamiltonian.sx>; }\n")
+            f.write("initialGuess { include <initialGuess.sx>; }\n")
             if self._generic_input["fix_spin_constraint"]:
-                self.input_writer.write_spins_constraints(
-                    file_name="spins.in",
-                    cwd=self.working_directory,
-                    spins_str=self._spins_str,
-                )
-        self.input.write_file(file_name="userparameters.sx", cwd=self.working_directory)
-        self.input_writer.write_main(
-            file_name="input.sx",
-            cwd=self.working_directory,
-            main_str=self._main_str,
-            spin_constraint_enabled=self._generic_input["fix_spin_constraint"],
-            enable_kjxc=enable_kjxc,
-            job_name=self.job_name,
+                f.write("spinConstraint { include <spinConstraint.sx>; }\n")
+            f.write("main { include <main.sx>; }\n")
+
+    def write_group(self, group, file_name):
+        file_name = posixpath.join(self.working_directory, file_name)
+        with open(file_name, "w") as f:
+            f.write(group.to_sx_str())
+
+    def write_species(self, file_name):
+        file_name = posixpath.join(self.working_directory, file_name)
+        with open(file_name, "w") as f:
+            for i in range(len(self.input.species["species"])):
+                f.write("species {\n")
+                for k, v in self.input.species["species"][i].items():
+                    f.write(f"\t{k} = {v};\n")
+                f.write("}\n")
+
+    def write_structure(self, file_name):
+        file_name = posixpath.join(self.working_directory, file_name)
+        with open(file_name, "w") as f:
+            f.write(f"cell = {self.input.structure.cell};\n")
+            if "symmetry" in self.input.structure:
+                f.write("symmetry {\n")
+                f.write(self.input.structure.symmetry.to_sx_str(level=1))
+                f.write("}\n")
+            for species in self.input.structure["species"]:
+                f.write("species {\n")
+                elt = species["element"]
+                f.write(f"\telement = {elt};\n")
+                for atom in species["atom"]:
+                    f.write("\tatom {\n")
+                    f.write(Group(atom).to_sx_str(level=2))
+                    f.write("\t}\n")
+                f.write("}\n")
```

Comment:
Oh nice catch; yeah they do. I _really_ like that `odict_to_spx_str` function!

Since they're replaceable with `write_group` I removed `write_species` and `write_structure` in my latest commit